dockerfiles/nodejs-21.Dockerfile

```diff
@@ -0,0 +1,17 @@
+FROM node:21.7-alpine3.19
+
+ENV CODECRAFTERS_DEPENDENCY_FILE_PATHS="package.json,package-lock.json"
+
+WORKDIR /app
+
+COPY package.json ./
+COPY package-lock.json ./
+
+RUN npm i 
+
+RUN mkdir -p /app-cached
+# If the node_modules directory exists, move it to /app-cached
+RUN if [ -d "/app/node_modules" ]; then mv /app/node_modules /app-cached; fi
+
+RUN printf "cd \${CODECRAFTERS_SUBMISSION_DIR} && npm i" > /codecrafters-precompile.sh
```

Comment:
@ryan-gang we shouldn't need to `npm install` again, no? The only reason to npm install is if either `package.json` or `package-lock.json` have changed, and in those cases the whole docker image will get build again using this Dockerfile.